v2/internal/test/fake/bundleendpoint.go

```diff
@@ -0,0 +1,165 @@
+package fake
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/spiffebundle"
+	"github.com/spiffe/go-spiffe/v2/internal/test"
+	"github.com/spiffe/go-spiffe/v2/internal/x509util"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"github.com/stretchr/testify/assert"
+)
+
+type BundleEndpoint struct {
+	tb     testing.TB
+	wg     sync.WaitGroup
+	ctx    context.Context
+	port   int
+	server *http.Server
+	sm     *http.ServeMux
+	// Root certificates used by clients to verify server certificates.
+	rootCAs *x509.CertPool
+	// TLS configuration used by the server.
+	tlscfg *tls.Config
+	// SPIFFE bundle returned by this BundleEndpoint.
+	bundle *spiffebundle.Bundle
+}
+
+type BundleEndpointOption interface {
+	apply(*BundleEndpoint)
+}
+
+// BEOption groups the available options for a BundleEndpoint.
+var BEOption = new(bundleEndpointOption)
+
+func NewBundleEndpoint(ctx context.Context, tb testing.TB, port int, option ...BundleEndpointOption) *BundleEndpoint {
+	rootCAs, cert := test.CreateWebCredentials(tb)
+	tlscfg := &tls.Config{
+		Certificates: []tls.Certificate{*cert},
+	}
+
+	be := &BundleEndpoint{
+		tb:      tb,
+		wg:      sync.WaitGroup{},
+		ctx:     ctx,
+		port:    port,
+		sm:      http.NewServeMux(),
```

Comment:
instead of holding onto the serve mux, you can just initialize the new serve mux in this function and pass it directly as the handler to the `http.Server` (and then serveHTTP can be removed).